src/librustdoc/html/highlight.rs

```diff
@@ -440,12 +440,13 @@ impl<'a> Classifier<'a> {
                 .unwrap_or(false)
             {
                 let tokens = self.get_full_ident_path();
-                for (token, start, end) in &tokens {
-                    let text = &self.src[*start..*end];
-                    self.advance(*token, text, sink, *start as u32);
+                let is_empty = tokens.is_empty();
+                for (token, start, end) in tokens {
+                    let text = &self.src[start..end];
+                    self.advance(token, text, sink, start as u32);
                     self.byte_pos += text.len() as u32;
                 }
-                if !tokens.is_empty() {
+                if is_empty {
```

Comment:
Reversing this condition seems suspicious. Especially because `tokens` is not mutable, so calculating the empty state before or after the loop shouldn't make a difference. :thinking: 

I'm not familiar with rustdoc though, so maybe I just don't have enough insight into what this code does.

Reply:
Whoops! I will fix this shortly. Thanks for catching it.

Reply:
Fixed.